cmd/boost/provider_cmd_test.go

```diff
@@ -32,6 +33,11 @@ func TestMultiaddrToNative(t *testing.T) {
 		proto:    "fancynewproto",
 		ma:       "/dns/foo.com/tcp/80/http",
 		expected: "",
+	}, {
+		name:     "test protocol",
+		proto:    "http",
+		ma:       "/dns/foo.com/tcp/7777/http/p2p/12D3KooWHf7cahZvAVB36SGaVXc7fiVDoJdRJq42zDRcN2s2512h",
+		expected: "http://foo.com:7777",
```

Comment:
+1, with plain HTTP there is no `/p2p/{peerid}` – remove it to avoid causing clients to do wasteful work  (clients like Kubo will try to dial multiaddr, fail because this is plain HTTP and no libp2p, then will retry by doing DHT/peer routing lookup for other multiaddrs for the peerid, and probably fail again). 

In case you plan to ever do support libp2p over HTTP, see  https://github.com/libp2p/specs/pull/508 and https://github.com/libp2p/specs/pull/550 

cc @MarcoPolo for visibility (perhaps we should have it explicitly written down somewhere in multiaddr specs that if there is no libp2p there should be no `/p2p/peerid` suffix? obvious, but maybe needed?)